CHANGELOG.md

```diff
@@ -77,6 +77,10 @@ Current
     * Created `MemoizingDimensionMappingResultSetMapper` to support field transform use case
 
 ### Changed:
+- [LuceneSearchProvider will throw an exception if a thread spends too much time waiting on acquiring a lock](https://github.com/yahoo/fili/issues/870)
+  * Currently, if LuceneSearchProvider tries to acquire a lock it will wait forever until the lock is released. If the 
+  lock is erroneously never leaked the requesting thread will hang forever.
+  * The new behavior is to timeout after some amount of time, throw an error, and fail the query.
```

Comment:
What is done to clean up the bad lock? Will all subsequent queries hit this timeout until the system is rebooted?

Reply:
We decided not to try to be too smart and clean up the bad lock. This PR just causes an actual error to be thrown and explicitly exposes the locking issue, instead of letting the requesting thread hang forever. I think a thread releasing a lock is a semi-decidable problem so I'm not sure under what case we would want to force a thread to give up the lock.